### PR TITLE
[3.9] Pre Update Checker Error descriptions

### PR DESCRIPTION
**Pull request by Kev1n337**
Pull Request for Issue #17820 .

### Summary of Changes

When the user wants to update to a new Joomla version but has a PHP version installed or uses a database type (sqlsrv or sqlazure when updating to Joomla 4.0) that is not supported by the new version, new error messages are shown to the user. Previously, in all of these cases, the notice "We can't find a download URL" was shown. This hint is now only shown, when the user has a compatible php version installed and uses a compatible DB type. Therefore, the JUpdate class now always returns the minimum php version, even when the currently installed php version is not compatible.

### Testing Instructions

1. Create file `list.xml` at `/`
```xml
<extensionset>
    <extension element="joomla" type="file" version="4.0.0" targetplatformversion="3.9" detailsurl="http://localhost/extension_sts.xml" />
</extensionset>
```
 
2. Create file `extension_sts.xml` at `/`
```xml
<updates>
    <update>
        <name>Joomla! 3.7</name>
        <description>Joomla! 3.7 CMS</description>
        <element>joomla</element>
        <type>file</type>
        <version>4.0.0</version>
        <infourl title="Joomla!">https://www.joomla.org/announcements/release-news/5711-joomla-3-7-5-release.html</infourl>
        <downloads>
            <downloadurl type="full" format="zip">https://downloads.joomla.org/cms/joomla3/3-7-5/Joomla_3.7.5-Stable-Update_Package.zip</downloadurl>
        </downloads>
        <tags>
            <tag>stable</tag>
        </tags>
        <maintainer>Joomla! PLT</maintainer>
        <maintainerurl>https://www.joomla.org</maintainerurl>
        <section>STS</section>
        <targetplatform name="joomla" version="3.[23456789]" />
        <php_minimum>7.0</php_minimum>
    </update>
</updates>
```

3. Components -> Joomla! Update -> Options
 
Custom URL: `localhost/list.xml`

4. Save & Close
 
There should be a fake update to Joomla 4.0 and you should see the pre update checker from #17742

5. When changing the version in <php_minimum>-tag to 8 in the file `extension_sts.xml` at `/` you should see a hint that your current php version is not supported.

6. Add your current database type (e.g. `mysqli`) to `$unsupportedDatabaseTypes` in function `isDatabaseTypeSupported()` in file `./administrator/components/com_joomlaupdate/models/default.php`.

7. You should now see two error descriptions, one for the not supported php version and one for the used db type being not compatible.

updateerrors

### Expected result

You see different error descriptions depending on the error preventing the update to be installed. (Wrong PHP Version, wrong DB type, problem with update server).

### Actual result

You only see one error description, no matter if the download fails due to a problem with the Joomla Update Server or due to your server not meeting the minimum update requirements.

### Documentation Changes Required

None.

### CODE REVIEW

File: administrator/language/en-GB/en-GB.com_joomlaupdate.ini [hunk 1]
```diff
@@ -34,7 +34,11 @@ COM_JOOMLAUPDATE_VIEW_COMPLETE_HEADING="Joomla Version Update Status"
...
-COM_JOOMLAUPDATE_VIEW_DEFAULT_NO_DOWNLOAD_URL_DESC="An update to Joomla %1$s was found, but it wasn't possible to fetch the download URL for that update. There are two possibilities why this happens:<br>- Your host doesn't support <a href="_QQ_"https://downloads.joomla.org/technical-requirements"_QQ_">the minimum requirements for Joomla %1$s</a> and there is no alternative download for your configuration available.<br>- There is a problem with the Joomla Update Server.<br><br>Please try to download the update package from <a href="_QQ_"https://downloads.joomla.org/latest"_QQ_">the official Joomla download page</a> and use the Upload and Update tab."
+COM_JOOMLAUPDATE_VIEW_DEFAULT_NO_DOWNLOAD_URL_DESC="An update to Joomla %1$s was found, but it wasn't possible to fetch the download URL for that update. There is a problem with the Joomla Update Server.<br><br>Please try to download the update package from <a href="_QQ_"https://downloads.joomla.org/latest"_QQ_">the official Joomla download page</a> and use the Upload and Update tab."
+COM_JOOMLAUPDATE_VIEW_DEFAULT_DB_NOT_SUPPORTED="Your Database type is not supported"
```
Comment (SamuelSchepp) [inline administrator/language/en-GB/en-GB.com_joomlaupdate.ini]: "Your database type"...

**Comment by SamuelSchepp** [comment]
I have tested this item :white_check_mark: successfully on 75e05751fb064180ca629300a715ecd797311535<br /><br />Tested 3 cases: unsupported php version, unsupported database type, both.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17849">issues.joomla.org/tracker/joomla-cms/17849</a>.</sub>

File: administrator/language/en-GB/en-GB.com_joomlaupdate.ini [hunk 2]
```diff
@@ -34,7 +34,11 @@ COM_JOOMLAUPDATE_VIEW_COMPLETE_HEADING="Joomla Version Update Status"
...
+COM_JOOMLAUPDATE_VIEW_DEFAULT_DB_NOT_SUPPORTED_DESC="An update to Joomla %1$s was found, but your current database type is not supported by the new version.<br> For further details take a look at <a href="_QQ_"https://downloads.joomla.org/technical-requirements"_QQ_">the minimum requirements for Joomla %1$s</a>."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_joomlaupdate.ini]: remove the space after the br

File: administrator/language/en-GB/en-GB.com_joomlaupdate.ini [hunk 3]
```diff
@@ -34,7 +34,11 @@ COM_JOOMLAUPDATE_VIEW_COMPLETE_HEADING="Joomla Version Update Status"
...
+COM_JOOMLAUPDATE_VIEW_DEFAULT_PHP_VERSION_NOT_SUPPORTED="Your PHP version is not supported"
+COM_JOOMLAUPDATE_VIEW_DEFAULT_PHP_VERSION_NOT_SUPPORTED_DESC="An update to Joomla %1$s was found, but your currently installed PHP version does not match <a href="_QQ_"https://downloads.joomla.org/technical-requirements"_QQ_">the minimum requirements for Joomla %1$s</a>"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_joomlaupdate.ini]: Please add a period (.) at the end of the sentence

**Comment by Schmidie64** [comment]
I have tested this item :white_check_mark: successfully on ff7abaa325e2a14ed3b16e214eaadfb903d81230<br /><br />@icampus
I could reproduce the Bug by following the testing Instructions. The item fixed the Bug. Everything works fine.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17849">issues.joomla.org/tracker/joomla-cms/17849</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Comment by Dennissehrt** [comment]
I have tested this item :white_check_mark: successfully on ff7abaa325e2a14ed3b16e214eaadfb903d81230<br /><br />
@icampus
I followed the test instruction and it was successfully<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17849">issues.joomla.org/tracker/joomla-cms/17849</a>.</sub>

**Comment by brianteeman** [comment]
Looks good - thanks

**Pull request merged**